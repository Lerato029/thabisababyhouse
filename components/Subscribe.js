/* ==================================================The Subscribe Component================================== */
/* importing react and react hooks */
import React, { useRef, useState } from "react";
//using the postData module to make post req
import { postData } from "../utils/fetchData";


const Subscribe = () => {
  //using hook to get email input value quicker
  const email = useRef(null);
  
  //message to be displayed to user
  const [message, setMessage] = useState("");

  /* ==========================================================Main Functionality */
  //===========================================================Create Subscription
  const subscribe = async (e) => {
    e.preventDefault();

    /* POST request to api with body object */
    const res = await postData("subscribe", { email: email.current.value });

    //check if an error is returned
    if (res.err) {
      //message to be read to user
      setMessage(err);
      return;
    }

    //else we success and reset email value
    email.current.value = "";
    //success msg to client
    setMessage(res.msg);
  };

  /* JSX Featuring a form to subscribe to mailing list*/
  return (
    <form onSubmit={subscribe}>
      <h2>Subscribe to our newsletter</h2>
      <div className="mb-3">
        <label htmlFor="exampleInputEmail1" className="form-label">
          Email address
        </label>
        <input
          type="email"
          className="form-control"
          id="exampleInputEmail1"
          aria-describedby="emailHelp"
          required
          ref={email}
        />
        <div id="emailHelp" className="form-text">
          We'll never share your email with anyone else.
        </div>
      </div>
      <div>{message ? message : `Get updated on the baby house`}</div>

      <button type="submit" className="btn btn-success">
        Subscribe
      </button>
    </form>
  );
};

export default Subscribe;
