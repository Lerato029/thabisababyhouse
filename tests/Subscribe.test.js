//============================Component Snapshot test
import React from "react";

//import to render react component
import renderer from "react-test-renderer";

//component with subscription form
import Subscribe from "../components/Subscribe";

//snapshot test
it("renders correctly", () => {
  const tree = renderer.create(<Subscribe />).toJSON();
  //compare with snapshot
  expect(tree).toMatchSnapshot();
});
