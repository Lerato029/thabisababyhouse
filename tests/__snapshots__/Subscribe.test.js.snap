// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <h2>
    Subscribe to our newsletter
  </h2>
  <div
    className="mb-3"
  >
    <label
      className="form-label"
      htmlFor="exampleInputEmail1"
    >
      Email address
    </label>
    <input
      aria-describedby="emailHelp"
      className="form-control"
      id="exampleInputEmail1"
      type="email"
    />
    <div
      className="form-text"
      id="emailHelp"
    >
      We'll never share your email with anyone else.
    </div>
  </div>
  <div>
    Get updated on the baby house
  </div>
  <button
    className="btn btn-success"
    type="submit"
  >
    Subscribe
  </button>
</form>
`;
